CodePush.js

```diff
@@ -76,45 +78,36 @@ async function checkForUpdate(deploymentKey = null, handleBinaryVersionMismatchC
            */
           const runtimeVersion = sharedCodePushOptions.runtimeVersion;
 
+          const versioningMode = sharedCodePushOptions.versioningMode;
+          const Versioning = (() => {
+            if (versioningMode === 'semver') {
+              return SemverVersioning;
+            }
+            if (versioningMode === 'incremental') {
+              return IncrementalVersioning;
+            }
+            if (versioningMode === 'custom' && sharedCodePushOptions.customVersioning) {
+              return sharedCodePushOptions.customVersioning
+            }
 
-          // NOTE: It is not necessary to implement it in three steps. However, it is recommended to implement unit-test-friendly.
-          /**
-           * TODO 1: find latest release in releaseHistory
-           * @return {ReleaseInfo}
-           */
-          function findLatestRelease(releaseHistory) {}
-
-          /**
-           * TODO 2: check if the update is mandatory
-           * @return {boolean}
-           */
-          function checkIsMandatory(runtimeVersion, releaseHistory) {}
-
-          /**
-           * TODO 3: determine whether to rollback and execute it
-           * @return {boolean}
-           */
-          function shouldRollback(runtimeVersion, releaseHistory) {}
-          if (shouldRollback(runtimeVersion, releaseHistory)) {
-            // rollback
-          }
-
-          // NOTE: sample code
-          const update = findLatestRelease(releaseHistory);
-          const isMandatory = checkIsMandatory(runtimeVersion, update);
+            throw new Error('No versioning object is defined');
+          })()
```

Comment:
@changwoolab 
If we declare a base class and implement subclasses with `findLatestRelease`, `checkIsMandatory`, and `shouldRollback`, wouldn’t it make better use of polymorphism?
Of course, it seems unlikely we’ll add more versioning methods beyond semver, incremental, and custom.

What do you think about it?

Reply:
@floydkim 
Included in [e655ebf](https://github.com/Soomgo-Mobile/react-native-code-push/pull/27/commits/e655ebff78302de0bcfaaf5d666c18a4f8021553)